sstables/sstables.hh

```diff
@@ -417,6 +417,17 @@ public:
 
     bool requires_view_building() const noexcept { return _state == sstable_state::staging; }
 
+    future<> wait_until_view_is_built() {
+        // wait for sstable to move out of staging, which is done after the view is updated.
+        if (_state != sstable_state::staging) {
+            return make_ready_future();
+        }
+
+        return _state_changed_cond.wait([this] {
+            return _state != sstable_state::staging;
+        });
+    }
```

Comment:
I don't like this logic being internalized to the sstable. We should avoid making sstable start to collect pieces of logic that really belong to higher-level layers that use the sstable.

Reply:
Ack. New version changes the approach.